admin/timescale/reset_tables.sql

```diff
@@ -0,0 +1,9 @@
+BEGIN;
+
+DELETE FROM listen                  CASCADE;
```

Comment:
Only for large tables, it seems:

https://www.lob.com/blog/truncate-vs-delete-efficiently-clearing-data-from-a-postgres-table#:~:text='TRUNCATE'%20is%20a%20fast%20operation,inserts%20a%20handful%20of%20rows.

I guess in the context of tests it will always be small tables, so I guess this is fine.

Reply:
FWIW, The first solution I had tried was to use TRUNCATE  to replace CREATE/DROP and found that there is noticeable speed improvement. Disappointed by the results, I formed the belief that we shouldn't do any table clearing at all in the tests and move it all transactions and reverse those transactions in ROLLBACK. That was the approach I tried out in https://github.com/metabrainz/listenbrainz-server/pull/2094 and it produced great speedups.

However, the changes there are too much so I was on the lookout for a less intrusive way. Randomly I tried DELETE FROM and observed the speed up. If only I had found the blog post you linked above earlier!!

DELETE FROM speeds up the test to a good extent already, but we can still try using transactional tests later if those provide further speed up or other reasons.